core/test/utilities/helpers/navigation-helpers.spec.js

```diff
@@ -2,9 +2,93 @@
 const chai = require('chai');
 const assert = chai.assert;
 const sinon = require('sinon');
-import { NavigationHelpers } from '../../../src/utilities/helpers';
+import { AuthHelpers, NavigationHelpers } from '../../../src/utilities/helpers';
+import { LuigiAuth, LuigiConfig } from '../../../src/core-api';
 
 describe('Navigation-helpers', () => {
+  describe('isNodeAccessPermitted', () => {
+    beforeEach(() => {
+      sinon.stub(LuigiAuth, 'isAuthorizationEnabled');
+      sinon.stub(AuthHelpers, 'isLoggedIn');
+      sinon.stub(LuigiConfig, 'getConfigValue');
+    });
+    afterEach(() => {
+      sinon.restore();
+    });
+
+    it('without permissionCheckerFn', () => {
+      LuigiConfig.getConfigValue.returns(null);
+      assert.isTrue(NavigationHelpers.isNodeAccessPermitted());
+    });
+
+    it('logged in, normal anonymousAccess', () => {
+      const checkNode = {
+        anonymousAccess: true
+      };
+      const permissionCheckerFn = sinon.spy();
+
+      LuigiAuth.isAuthorizationEnabled.returns(true);
+      AuthHelpers.isLoggedIn.returns(true);
+
+      assert.isTrue(NavigationHelpers.isNodeAccessPermitted(checkNode));
+    });
+
+    it('logged out, normal anonymousAccess', () => {
+      const checkNode = {
+        anonymousAccess: true
+      };
+
+      LuigiAuth.isAuthorizationEnabled.returns(true);
+      AuthHelpers.isLoggedIn.returns(false);
+
+      assert.isTrue(NavigationHelpers.isNodeAccessPermitted(checkNode));
+    });
+
+    it('logged in, exclusive anonymousAccess', () => {
+      const checkNode = {
+        anonymousAccess: 'exclusive'
+      };
+      const permissionCheckerFn = sinon.spy();
+
+      LuigiAuth.isAuthorizationEnabled.returns(true);
+      AuthHelpers.isLoggedIn.returns(true);
+      LuigiConfig.getConfigValue.returns(permissionCheckerFn);
+
+      assert.isFalse(NavigationHelpers.isNodeAccessPermitted(checkNode));
+    });
+
+    it('logged out, exclusive anonymousAccess', () => {
```

Comment:
Please update the mocks for this tests, as they are the same as in the test above

Reply:
Done, we need to check if the exclusive anonymous access implementation is valid, since it also calls the permission checker function.

Reply:
Please bring this topic to the next Scrum meeting ;)